resources/aws/errors.go

```diff
@@ -64,24 +52,76 @@ func IsRouteNotFoundError(err error) bool {
 	return errgo.Cause(err) == routeNotFoundError
 }
 
-func IsSecurityGroupFind(err error) bool {
-	return errgo.Cause(err) == securityGroupFindError
+var securityGroupNotFoundError = errgo.New("security group not found")
+
+// IsSecurityGroupNotFound asserts securityGroupNotFoundError.
+func IsSecurityGroupNotFound(err error) bool {
+	return errgo.Cause(err) == securityGroupNotFoundError
 }
 
-func IsSubnetFind(err error) bool {
-	return errgo.Cause(err) == subnetFindError
+var subnetNotFoundError = errgo.New("subnet not found")
+
+// IsSubnetNotFound asserts subnetNotFoundError.
+func IsSubnetNotFound(err error) bool {
+	return errgo.Cause(err) == subnetNotFoundError
+}
+
+var vpcNotFoundError = errgo.New("vpc not found")
+
+// IsVpcNotFound asserts vpcNotFoundError.
+func IsVpcNotFound(err error) bool {
+	return errgo.Cause(err) == vpcNotFoundError
 }
 
-func IsVpcFindError(err error) bool {
-	return errgo.Cause(err) == vpcFindError
+// Delete errors.
+
+var resourceDeleteError = errgo.New("Couldn't delete resource, it lacks the necessary data (ID)")
+
+// IsResourceDelete asserts resourceDeleteError.
+func IsResourceDelete(err error) bool {
+	return errgo.Cause(err) == resourceDeleteError
 }
 
+// Other errors.
+
+var clientNotInitializedError = errgo.New("The client has not been initialized")
+
+// IsClientNotInitialized asserts clientNotInitializedError.
+func IsClientNotInitialized(err error) bool {
+	return errgo.Cause(err) == clientNotInitializedError
+}
+
+var keyPairCannotCreateAndNotFoundError = errgo.New("Couldn't create and find the keypair")
+
+// IsKeyPairCannotCreateAndNotFound asserts keyPairCannotCreateAndNotFoundError.
+func IsKeyPairCannotCreateAndNotFound(err error) bool {
```

Comment:
I think this should be two different errors which are handled separately. Having `And` in the name smells.